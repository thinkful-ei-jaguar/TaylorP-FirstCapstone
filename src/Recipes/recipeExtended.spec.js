import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import RecipeExtended from './recipeExtended'
import RecipeContext from '../Context/recipeContext'

describe('<RecipeExtended />', () => {
  it('renders without crashing', () => {
    const match = {params: {id: 1}}
    mount(<MemoryRouter><RecipeExtended match={match}/></MemoryRouter>)
  });
  it('renders Recipe Extended with recipe info', () => {
    const match = {params: {id: 1}}
    const ctxval = {
      recipe: {
          id: 1,
          recipe_name: 'test-recipe',
          recipe_ingredients: ['step1','step2','step3'],
          recipe_prep: 'do the thing and test it',
          recipe_img: null
        },
      favoriteList: [],
      spirit: 'all',
      error: null,
      setError: () => {},
      clearError: ()=> {},
      setRecipe: () => {},
      clearRecipe: ()=> {},
      setFavoriteList: () => {},
      setSpirit: () => {},
      addFavorite: () => {}
    }
    const tree = renderer.create(
      <MemoryRouter>
        <RecipeContext.Provider value={ctxval}>
          <RecipeExtended match={match} />
        </RecipeContext.Provider>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})