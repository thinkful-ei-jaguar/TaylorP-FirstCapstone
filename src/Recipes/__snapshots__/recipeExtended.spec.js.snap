// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeExtended /> renders Recipe Extended with recipe info 1`] = `
<main
  className="expanded-main"
>
  <section
    className="recipe-expanded"
  >
    <div
      className="recipe-name-icon-div"
    >
      <h2>
        test-recipe
      </h2>
    </div>
    <div
      className="recipe-specs-make-div"
    >
      <div
        className="specs-div"
      >
        <h3>
          Ingredients:
        </h3>
        <ul>
          <li>
            step1
          </li>
          <li>
            step2
          </li>
          <li>
            step3
          </li>
        </ul>
      </div>
      <div
        className="make-div"
      >
        <h3>
          Steps:
        </h3>
        <p>
          do the thing and test it
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Add to Favorites
        </button>
      </div>
    </div>
  </section>
  <button
    className="go-back"
    onClick={[Function]}
  >
    Go Back
  </button>
</main>
`;
